hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -88,61 +87,72 @@ public ContractMetrics(
     }
 
-    public void createContractMetrics() {
-
-        final var contractsConfig = requireNonNull(contractsConfigSupplier.get());
-        this.p1MetricsEnabled = contractsConfig.metricsSmartContractPrimaryEnabled();
-        this.p2MetricsEnabled = contractsConfig.metricsSmartContractSecondaryEnabled();
+    private final Object metricsCreationLock = new Object();
+    private volatile boolean primaryMetricsAreCreated = false;
 
-        final var metrics = requireNonNull(metricsSupplier.get());
-
-        if (p1MetricsEnabled) {
-            // Rejected transactions counters
-            for (final var txKind : POSSIBLE_FAILING_TX_TYPES.keySet()) {
-                final var name = toRejectedName(txKind, REJECTED_TXN_SHORT_DESCR);
-                final var descr = toRejectedDescr(txKind, REJECTED_TXN_SHORT_DESCR);
-                final var config = new Counter.Config(METRIC_CATEGORY, name)
-                        .withDescription(descr)
-                        .withUnit(METRIC_TXN_UNIT);
-                final var metric = newCounter(metrics, config);
-                rejectedTxsCounters.put(txKind, metric);
-            }
-
-            // Rejected transactions because they don't even have intrinsic gas
-            for (final var txKind : POSSIBLE_FAILING_TX_TYPES.keySet()) {
-                final var functionalityName = POSSIBLE_FAILING_TX_TYPES.get(txKind) + "DueToIntrinsicGas";
-                final var name = toRejectedName(functionalityName, REJECTED_FOR_GAS_SHORT_DESCR);
-                final var descr = toRejectedDescr(functionalityName, REJECTED_FOR_GAS_SHORT_DESCR);
-                final var config = new Counter.Config(METRIC_CATEGORY, name)
-                        .withDescription(descr)
-                        .withUnit(METRIC_TXN_UNIT);
-                final var metric = newCounter(metrics, config);
-                rejectedTxsLackingIntrinsicGas.put(txKind, metric);
-            }
+    public void createContractMetrics() {
 
-            // Rejected transactions for ethereum calls that are in type 3 blob transaction format
-            {
-                final var name = toRejectedName(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
-                final var descr = toRejectedDescr(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
-                final var config = new Counter.Config(METRIC_CATEGORY, name)
-                        .withDescription(descr)
-                        .withUnit(METRIC_TXN_UNIT);
-                final var metric = newCounter(metrics, config);
-                rejectedEthType3Counter = metric;
+        // Primary metrics are a fixed set and can be created when `Hedera` initializes the system.
+        // But it actually must wait until the platform calls `Hedera.onStateInitialized` and,
+        // remarkably, that can happen more than once.  We must only create the metrics the
```

Comment:
There is a pretty mundane explanation for this; if a node's network connection is interrupted and then restored, its reconnect phase will happen within the same JVM; and will hence re-initialize the same `Hedera` singleton.

Reply:
yes, i finally figured that out.  but in fact, in the P2 metrics PR (coming shortly) I figured out the _correct_ way to handle this using InitTrigger.GENESIS, so all this garbage is out.